pxtcompiler/emitter/service.ts

```diff
@@ -955,29 +988,47 @@ namespace ts.pxtc.service {
                 const propertyAccessTarget = findInnerMostNodeAtPosition(tsAst, isPython ? tsPos : dotIdx - 1)
 
                 if (propertyAccessTarget) {
-                    let type: Type;
-
                     const symbol = tc.getSymbolAtLocation(propertyAccessTarget);
-                    if (symbol) {
-                        type = tc.getTypeOfSymbolAtLocation(symbol, propertyAccessTarget);
-                    }
-                    else {
-                        type = tc.getTypeAtLocation(propertyAccessTarget);
-                    }
+                    if (symbol?.members?.size > 0) {
```

Comment:
Instead of separate handling for `this`, could you just call `tc.getPropertiesOfType()` on the type object? You might have to check to see if that works on `this` types but it would unify the two clauses

Reply:
Yeah from what I could tell this didn't really work with "this". If you get the symbol for a location with text "this" it returns the type symbol, which is strange, I would have expect a "this" symbol and then checking the type at that location would get the type.

I can take another look, maybe I can unify handling of "this".

Reply:
Figured it out. The missing piece was you need to use "getTypeOfSymbolAtLocation" to get the type of symbols for instances of "this", whereas in all other cases you use "getTypeOfSymbolAtLocation".

I still seems a bit odd to me, like they skipped a step. The symbol for "this" should be "this" and getting the type of that symbol at that location should be "Foo". But whatever.